### PR TITLE
IAM-318-COS-Integration

### PR DESCRIPTION
**Pull request by bencekov**
Integrating Identity Platform Login UI charm with COS components: Prometheus and Loki

cos integrations manual testing instructions:

These instructions presuppose that you have a juju cluster with working prometheus-k8s, and loki-k8s integrated with login UI. Login UI charm uses the default relation names.

Prometheus:
curl 'prometheus_url:9090/api/v1/query?query=http_requests_total&app=identity-platform-login-ui-operator&code=200&endpoint=/admin/health/alive' | yq
If you get a list of metrics, then it worked.

Loki
curl loki_url:3100/loki/api/v1/label/juju_unit/values
If your unit shows up in the results, it should be good.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -137,7 +180,10 @@ def _login_ui_layer(self) -> Layer:
                 self._container_name: {
                     "override": "replace",
                     "summary": "identity platform login ui",
-                    "command": "identity_platform_login_ui",
+                    "command": '/bin/sh -c "{} 2>&1 | tee -a {}"'.format(
+                        self._login_ui_service_command,
+                        str(self._log_path),
+                    ),
```
Comment (nsklikas) [inline src/charm.py]: Thought: Would it make sense to solve the log forwarding problem on the login-ui rather than using this hack?

Comment (bencekov) [reply src/charm.py:186]: For this, I would review the other Loki Consumer object in the loki charm lib. The current we use which is LogProxyConsumer downloads a promtail binary, and configures it with the given log files.

Comment (bencekov) [reply src/charm.py]: If we all agree on having the login UI write the logs into a file, I'll just change this.

**Comment by natalian98** [comment]
@bencekov Could you check why the CI is failing? Does this PR depend on any other PR in identity-platform-login-ui?

**Comment by natalian98** [comment]
@bencekov I think once you rebase your branch, the CI should pass.

**Review by shipperizer — APPROVED** [review]

**Pull request merged**